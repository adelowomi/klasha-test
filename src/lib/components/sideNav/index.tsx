import { ArrowLeftIcon } from "@chakra-ui/icons";
import {
  Box,
  BoxProps,
  Button,
  ComponentWithAs,
  Flex,
  Icon,
  IconProps,
  Text,
} from "@chakra-ui/react";
import { menuItems, menuItems2, menuItems3 } from "data/SideNavData";

import * as React from "react";
import { IDisclosureProps } from "../appLayout";
import { LogoComponent } from "../assets";
import { SideNavItem } from "./sideNavItems";

export interface ISideNavProps extends BoxProps {
  currentRoute: string;
  disclosure: IDisclosureProps;
}

export interface IMenuItem {
  name: string;
  route: string;
  icon: ComponentWithAs<"svg", IconProps>;
}

export const SideNav: React.FC<ISideNavProps> = ({
  currentRoute,
  disclosure,
  ...rest
}) => {
  return (
    <Flex
      minH={"100vh"}
      direction="column"
      w={{ base: "full", md: "280px" }}
      bg={"brand.400"}
      borderColor={"brand.400"}
      justifyContent={"space-between"}
      {...rest}
    >
      <Box>
        <Flex direction={"column"} pt={"12"} pl={"12"}>
          <a href="/">
            <LogoComponent fill={"black"} />
          </a>
        </Flex>
        <Flex direction={"column"} mt={"10"} pl={"12"}>
          <Text fontWeight="bold" fontSize={"medium"} color="brand.100">
            Main pages
          </Text>
          {menuItems.map((item, index) => (
            <SideNavItem
              key={index}
              name={item.name}
              route={item.route}
              currentRoute={currentRoute}
            >
              <Icon
                as={item.icon}
                _hover={{
                  color: `${
                    currentRoute == `${item.route}` ? "white" : "brand.200"
                  }`,
                }}
                color={`${
                  currentRoute == item.route ? "brand.200" : "brand.300"
                }`}
              />
            </SideNavItem>
          ))}
        </Flex>
        <Flex direction={"column"} mt={"10"} pl={"12"}>
          <Text fontWeight="bold" fontSize={"medium"} color="brand.100">
            Accept Payments
          </Text>
          {menuItems2.map((item, index) => (
            <SideNavItem
              key={index}
              name={item.name}
              route={item.route}
              currentRoute={currentRoute}
            >
              <Icon
                as={item.icon}
                color={`${
                  currentRoute == item.route ? "brand.200" : "brand.300"
                }`}
              />
              {/* <FiGrid
                            color={`${
                                currentRoute == "dashboard"
                                    ? "#EF2C5A"
                                    : "#000000"
                            }`}
                            size={"24px"}
                        /> */}
            </SideNavItem>
          ))}
        </Flex>
        <Flex direction={"column"} mt={"10"} pl={"12"}>
          <Text fontWeight="bold" fontSize={"medium"} color="brand.100">
            Send Payments
          </Text>
          {menuItems3.map((item, index) => (
            <SideNavItem
              key={index}
              name={item.name}
              route={item.route}
              currentRoute={currentRoute}
            >
              <Icon
                as={item.icon}
                color={`${
                  currentRoute == item.route ? "brand.200" : "brand.300"
                }`}
              />
              {/* <FiGrid
                            color={`${
                                currentRoute == "dashboard"
                                    ? "#EF2C5A"
                                    : "#000000"
                            }`}
                            size={"24px"}
                        /> */}
            </SideNavItem>
          ))}
        </Flex>
      </Box>
      <Box
        pb={"12"}
        pl={"12"}
        justifySelf={"end"}
        display={{ base: "flex", md: "flex", lg: "none" }}
      >
        <Button
          leftIcon={<ArrowLeftIcon />}
          colorScheme="brand"
          width="148px"
          variant="outline"
          borderColor="brand.300"
          color={"brand.300"}
          fontSize="12px"
          onClick={() => disclosure.onClose()}
          _hover={{
            color: "white",
            backgroundColor: "brand.200",
            borderColor: "white",
          }}
        >
          Hide Panel
        </Button>
      </Box>
    </Flex>
  );
};
