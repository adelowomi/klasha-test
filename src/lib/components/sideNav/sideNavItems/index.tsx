import { Box, Flex, Text, Link } from "@chakra-ui/react";
import NextLink from "next/link";
import { FC } from "react";
export interface ISideNavItem {
  route: string;
  currentRoute: string;
  name: string;
  children?: React.ReactNode;
}
export const SideNavItem: FC<ISideNavItem> = ({
  children,
  currentRoute,
  name,
  route,
}) => (
  <NextLink href={`/${route}`} passHref>
    {/* <Link _focus={{ outline: "none" }}> */}
    <Flex
      data-testid={`side-nav-item-${route}`}
      marginBottom={3}
      // bgColor={currentRoute == `${route}` ? "brand.100" : ""}
      // pl={6}
      borderRightRadius={"3xl"}
      position={"relative"}
      h={"12"}
      color={currentRoute == `${route}` ? "brand.200" : "brand.300"}
      _hover={{
        bg: `${currentRoute == `${route}` ? "white" : "blackAlpha.50"}`,
        bgColor: `${currentRoute == `${route}` ? "brand.200" : ""}`,
        color: `${currentRoute == `${route}` ? "white" : "brand.200"}`,
      }}
    >
      <Flex alignItems={"center"}>
        {children}
        <Box ml={"3"}>
          <Text
            as={"span"}
            // color={currentRoute == `${route}` ? "brand.200" : "brand.300"}
            // _hover={{
            //   color: `${currentRoute == `${route}` ? "white" : "brand.200"}`,
            // }}
            cursor={"pointer"}
          >
            {`${name}`}
          </Text>
        </Box>
      </Flex>
    </Flex>
    {/* </Link> */}
  </NextLink>
);
