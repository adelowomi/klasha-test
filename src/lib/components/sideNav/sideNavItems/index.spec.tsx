import { render } from "@test";
import { SideNavItem } from "@components/sideNav/sideNavItems";
import { FiGrid } from "react-icons/fi";
import "@testing-library/jest-dom";

describe("SideNavItem", () => {
    it("should match snapshot", async () => {
        const currentRoute = "dashboard";
        const { asFragment } = render(
            <SideNavItem
                name="Dashboard"
                route="dashboard"
                currentRoute={currentRoute}
            >
                <FiGrid
                    color={`${
                        currentRoute == "dashboard" ? "#FFFFFF" : "#767171"
                    }`}
                    size={"24px"}
                />
            </SideNavItem>,
        );
        expect(asFragment()).toMatchSnapshot();
    });

    it("should render SideNavBar", async () => {
        const currentRoute = "dashboard";
        const sideNavBar = render(
            <SideNavItem
                name="Dashboard"
                route="dashboard"
                currentRoute={currentRoute}
            >
                <FiGrid
                    color={`${
                        currentRoute == "dashboard" ? "#FFFFFF" : "#767171"
                    }`}
                    size={"24px"}
                />
            </SideNavItem>,
        );
        expect(sideNavBar).toBeTruthy();
    });

    it("should have background color brand.200", async () => {
        const currentRoute = "dashboard";
        const { getByTestId, getByText } = render(
            <SideNavItem
                name="Dashboard"
                route="dashboard"
                currentRoute={currentRoute}
            >
                <FiGrid
                    color={`${
                        currentRoute == "dashboard" ? "#FFFFFF" : "#767171"
                    }`}
                    size={"24px"}
                />
            </SideNavItem>,
        );
        const item = getByTestId("side-nav-item-dashboard");
        // expect(item).toHaveTextContent("Dashboard");
        expect(getByText("Dashboard")).toBeInTheDocument();
        expect(item).toHaveStyle({ color: { r: 239, g: 44, b: 90 } });
    });
});
