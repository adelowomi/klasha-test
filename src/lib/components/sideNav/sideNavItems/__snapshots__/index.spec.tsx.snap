// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SideNavItem should match snapshot 1`] = `
<DocumentFragment>
  <a
    class="chakra-link css-1udx80"
    href="/dashboard"
  >
    <div
      class="css-dmwhn"
      data-testid="side-nav-item-dashboard"
    >
      <div
        class="css-70qvj9"
      >
        <svg
          color="#FFFFFF"
          fill="none"
          height="24px"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          style="color: rgb(255, 255, 255);"
          viewBox="0 0 24 24"
          width="24px"
          xmlns="http://www.w3.org/2000/svg"
        >
          <rect
            height="7"
            width="7"
            x="3"
            y="3"
          />
          <rect
            height="7"
            width="7"
            x="14"
            y="3"
          />
          <rect
            height="7"
            width="7"
            x="14"
            y="14"
          />
          <rect
            height="7"
            width="7"
            x="3"
            y="14"
          />
        </svg>
        <div
          class="css-1kw2fa0"
        >
          <span
            class="chakra-text css-qr814n"
          >
            Dashboard
          </span>
        </div>
      </div>
    </div>
  </a>
</DocumentFragment>
`;
